test/FluentMigrator.Tests/Unit/MigrationRunnerTests.cs

```diff
@@ -750,6 +753,24 @@ public void CanRunBreakingChangesInPreview()
                     new MigrationInfo(7, TransactionBehavior.Default, true, new TestBreakingMigration()), true));
         }
 
+        [Test]
+        public void CanLoadCustomMigrationScopeHandler()
+        {
+            _serviceCollection.AddScoped<IMigrationScopeHandler>(scoped => { return _migrationScopeHandlerMock.Object; });
+            var runner = CreateRunner();
+            runner.BeginScope();
+            _migrationScopeHandlerMock.Verify(x => x.BeginScope(), Times.Once());
+        }
+
+        [Test]
+        public void CanLoadDefaultMigrationScopeHandlerIfNoCustomHandlerIsSpecified()
+        {
+            var runner = CreateRunner();
+            BindingFlags bindFlags = BindingFlags.Instance | BindingFlags.NonPublic;
+            FieldInfo field = typeof(MigrationRunner).GetField("_migrationScopeHandler", bindFlags);
```

Comment:
I never used AutoFixture, im trying to quickly google now and it looks reasonable. I guess from contributor point of view - will be good to have few examples provided so everybody can use as a baseline for his tests.

Reply:
Yeah, a good overview is available here: https://github.com/bardock/Bardock.UnitTesting/blob/master/docs/talk/thirdparties_autofixture.md

The main thing I use it for is the Auto-Mocking Container design pattern, which I find saves a lot of code and also makes tests less brittle, since mock objects are generated by the Auto-Mocking Container.  https://github.com/jzabroski/Home/tree/master/Testing#auto-mocking-container-design-pattern

Reply:
@BigYellowHammer I'll close this sub-stream of comments.  It should go in a new ticket as its outside the scope of the original specification in #1047.  We'll leave the reflection logic as-is for now.

Reply:
I don't mind to use AutoFixture for this. 👍